packages/node_modules/@webex/test-users/src/whistler.js

```diff
@@ -0,0 +1,197 @@
+ */
+
+/**
+  * @typedef {Object} TestUserObject
+  * @property {string} password
+  * @property {string} emailAddress
+  * @property {string} displayName
+  */
+
+/**
+ * @typedef {Object} CreateUserOptions
+ * @param {Object} [options]
+ * @param {string} [options.clientId] defaults to WEBEX_CLIENT_ID
+ * @param {string} [options.clientSecret] defaults to WEBEX_CLIENT_SECRET
+ * @param {string} [options.conversationServiceUrl] defaults to WEBEX_TEST_USERS_CONVERSATION_SERVICE_URL
+ * @param {string} [options.idbrokerUrl] defaults to IDBROKER_BASE_URL
+ * @param {string} [options.orgId] organization ID to create the user under
+ * @param {string} [options.machineAccount] defaults to WHISTLER_MACHINE_ACCOUNT
+ * @param {string} [options.machinePassword] defaults to WHISTLER_MACHINE_PASSWORD
+ */
+
+/**
+ * Creates a test user
+ * @param {CreateUserOptions} options
+ * @returns {Promise.<TestUserObject>}
+ */
+export default function createTestUser(options = {}) {
+  const clientId = options.clientId || process.env.WEBEX_CLIENT_ID;
+  const clientSecret = options.clientSecret || process.env.WEBEX_CLIENT_SECRET;
+  const machineAccount = options.machineAccount || process.env.WHISTLER_MACHINE_ACCOUNT;
+  const machinePassword = options.idbrokerUrl || process.env.WHISTLER_MACHINE_PASSWORD;
+  const idbrokerUrl = options.idbrokerUrl || process.env.IDBROKER_BASE_URL;
+  const orgId = options.orgId || process.env.WHISTLER_TEST_ORG_ID;
+
+  if (!clientId) {
+    throw new Error('options.clientId or process.env.WEBEX_CLIENT_ID must be defined');
+  }
+
+  if (!clientSecret) {
+    throw new Error('options.clientSecret or process.env.WEBEX_CLIENT_SECRET must be defined');
+  }
+
+  if (!machineAccount) {
+    throw new Error('options.machineAccount or process.env.WHISTLER_MACHINE_ACCOUNT must be defined');
+  }
+
+  if (!machinePassword) {
+    throw new Error('options.machinePassword or process.env.WHISTLER_MACHINE_PASSWORD must be defined');
+  }
+  if (!idbrokerUrl) {
+    throw new Error('options.idbrokerUrl or process.env.IDBROKER_BASE_URL must be defined');
+  }
+
+  if (!orgId) {
+    throw new Error('options.orgId or process.env.WHISTLER_TEST_ORG_ID must be defined');
+  }
+
+  return getClientCredentials({
+    clientId,
+    clientSecret,
+    machineAccount,
+    machinePassword,
+    idbrokerUrl,
+    orgId
+  })
+    .then((authorization) => request({
+      method: 'GET',
+      uri: `${process.env.WHISTLER_API_SERVICE_URL}/reservations/testUser?isAccessTokenRequired=true`,
+      headers: {
+        authorization
+      }
+    }))
+    .then((res) => Object.assign({
+      password: res.body.responseMetaData.ciPassword,
+      emailAddress: res.body.responseMetaData.name,
+      displayName: res.body.responseMetaData.webExUserName,
+      token: res.body.responseMetaData.ciAccessToken,
+      reservationUrl: res.body.reservationUrl
+    }, res.body.responseMetaData));
+}
+
+/**
+ *
+ * @param {Object} options
```

Comment:
Can you document expected options?

Reply:
It's the entire user object so there's too much to document, also options isnt the best name, but i'm keeping it aligned with what's in `index.js`

Reply:
There is a type definition above for `TestUserObject`, if that is what it's supposed to be then, can you update the correct type?
Maybe rename the argument to `user`